arch/x86/core/x86_mmu.c

```diff
@@ -136,3 +136,155 @@ void _x86_mmu_set_flags(void *ptr, size_t size, u32_t flags, u32_t mask)
 		addr += MMU_PAGE_SIZE;
 	}
 }
+
+
+#ifdef CONFIG_MEMORY_DOMAIN
+
+/* Load the required parttions of the new incomming thread */
+void x86_load_mmu_mem_domain(struct k_thread *thread)
+{
+	u32_t partition_index;
+	u32_t total_partitions;
+	struct k_mem_partition partition;
+	struct k_mem_domain *mem_domain;
+	mem_domain = thread->mem_domain_info.mem_domain;
+
+	do {
+		/* If mem_domain doesn't point to a valid location return.*/
+		if (mem_domain == NULL) {
+			break;
+		}
+
+		/* Get the total number of partitions.
+		 * Block if the partitions exceed the max.*/
```

Comment:
I wouldn't use the word "Block" here. This comment is redundant anyway, it's pretty obvious what the next two lines do